crashtracker/src/collector/saguard.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+use nix::sys::signal::{self, SaFlags, SigAction, SigHandler};
+
+// Provides a lexically-scoped guard for signals
+// During execution of the signal handler, it cannot be guaranteed that the signal is handled
+// without SA_NODEFER, thus it also cannot be guaranteed that signals like SIGCHLD and SIGPIPE will
+// _not_ be emitted during this handler as a result of the handler itself. At the same time, it
+// isn't known whether it is safe to merely block all signals, as the user's own handler will be
+// given the chance to execute after ours. Thus, we need to prevent the emission of signals we
+// might create (and cannot be created during a signal handler except by our own execution) and
+// defer any other signals.
+// TODO this forces dynamic allocation in the signal handler
+pub struct SaGuard<const N: usize> {
+    old_sigactions: [(signal::Signal, signal::SigAction); N],
```

Comment:
is this going to use a bunch of our stack space?

Reply:
For the code we own we most likely are, but I don't think we can make that assumption for the stack walking.

Reply:
Yeah, stack walking is going to be broken in a bunch of ways.

FWIW, we recently broke the 8192 byte default limit in a different PR before unwinding ever happened 😭